pkg/apis/openebs.io/scp/v1alpha1/capabilties.go

```diff
@@ -14,112 +14,142 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-
 package v1alpha1
 
-// Capabilities to be supported by the pool or volume
+// Capabilities defines a set of attributes, which describe properties that the
+// storage pool or volume may support.
 type Capabilities struct {
-	// DataStorage capabilities
+	// DataStorage specifies accessModes, provisioning, multiPathing, compression
+	// and deduplication characteristics supported by a pool or a volume.
 	DataStorage *DataStorageCapabilities `json:"dataStorage"`
 
-	// Data security capabilities
-	// +nullable
+	// DataSecurity specifies security standards that is supported.
+	// +optional
 	DataSecurity *DataSecurityCapabilities `json:"dataSecurity,omitempty"`
 
-	// IO connectivity capabilities
-	// +nullable
+	// IOConnectivity describes capabilities to support various IO Connectivity options i.e accessProtocols
+	// +optional
 	IOConnectivity *IOConnectivityCapabilities `json:"IOConnectivity,omitempty"`
 
-	// IO performance capabilities
+	// IOPerformance describe the capabilities to support various IO performance options.
 	IOPerformance *IOPerformanceCapabilities `json:"IOPerformance"`
 
-	// DataProtection capabilities
-	// +nullable
+	// DataProtection describes data protection capabilities.
+	// +optional
 	DataProtection *DataProtectionCapabilities `json:"dataProtection,omitempty"`
 }
 
 // DataStorageCapabilities defines data storage capabilities for a pool or volume
 type DataStorageCapabilities struct {
-	//AccessModes contains the actual access modes
+	//AccessModes contains the actual access modes, viz ReadWriteOnce, ReadOnlyMany, ReadWriteMany
 	AccessModes []AccessMode `json:"accessModes"`
 
-	// ProvisioningPolicy defines provisioning policy type
-	ProvisioningPolicy []string `json:"provisioningPolicy"`
+	// ProvisioningPolicy defines provisioning policy type, viz thick, thin
+	ProvisioningPolicy []ProvisioningPolicy `json:"provisioningPolicy"`
 
 	// MultiPathing to be supported or not.
-	// +nullable
+	// +optional
 	MultiPathing []string `json:"multiPathing,omitempty"`
```

Comment:
This is a const as well - OnlineActive / OnlinePassive / None

Reply:
As none option is there the should it be not optional?